logic/tool/TypeResolver.java

```diff
@@ -373,88 +382,72 @@ private void convertValue(TypeVariable owner, ValueConstraint<?> constraint) {
+                //       Why do we not throw GraknException.of(ILLEGAL_STATE); ?
             }
         }
-
     }
 
-    private static class ResolverVariables {
+    private static class Resolvers {
 
-        private final Map<Identifier, TypeVariable> resolveeVars;
-        private final Map<RelationConstraint.RolePlayer, TypeVariable> rolePlayerResolveVars;
+        private final Map<Identifier, TypeVariable> typeResolvers;
+        private final Map<RelationConstraint.RolePlayer, TypeVariable> roleTypeResolvers;
+        // TODO: Why is there no checkers and getters for 'roleTypeResolvers'?
+        // TODO: do we still need 'roleTypeResolvers' given that we no longer put resolveRoleTypes on RelationConstraints?
 
-        private Integer tempCounter;
+        private int sysVarCounter;
 
-        ResolverVariables() {
-            this.tempCounter = 0;
-            this.resolveeVars = new HashMap<>();
-            this.rolePlayerResolveVars = new HashMap<>();
+        Resolvers() {
+            this.sysVarCounter = 0;
+            this.typeResolvers = new HashMap<>();
+            this.roleTypeResolvers = new HashMap<>();
         }
 
-        public TypeVariable convert(RelationConstraint.RolePlayer key) {
-            if (!rolePlayerResolveVars.containsKey(key)) {
-                TypeVariable newTypeVar = new TypeVariable(Identifier.Variable.of(
-                        new SystemReference("temp" + addAndGetCounter())));
-                rolePlayerResolveVars.put(key, newTypeVar);
-            }
-            return rolePlayerResolveVars.get(key);
+        TypeVariable register(RelationConstraint.RolePlayer rolePlayer) {
+            return roleTypeResolvers.computeIfAbsent(rolePlayer, rp -> new TypeVariable(newSystemVariable()));
         }
 
-        public TypeVariable convert(Variable key) {
-            if (!resolveeVars.containsKey(key.identifier())) {
+        TypeVariable register(Variable variable) {
+            return typeResolvers.computeIfAbsent(variable.identifier(), id -> {
                 TypeVariable newTypeVar;
-                if (key.reference().isAnonymous()) {
-                    newTypeVar = new TypeVariable(Identifier.Variable.of(
-                            new SystemReference("temp" + addAndGetCounter())));
-                } else newTypeVar = new TypeVariable(key.identifier());
-                if (key.isType()) {
-                    newTypeVar.copyConstraints(key.asType());
-                }
-                resolveeVars.put(key.identifier(), newTypeVar);
-            }
-            return resolveeVars.get(key.identifier());
+                if (variable.reference().isAnonymous()) newTypeVar = new TypeVariable(newSystemVariable());
+                else newTypeVar = new TypeVariable(variable.identifier());
+                if (variable.isType()) newTypeVar.copyConstraints(variable.asType());
+                return newTypeVar;
+            });
```

Comment:
The entire logic in the method body was equivalent to a computeIfAbsent() - we should simplify code as much as possible - the shorter the better, so we do not burden the reader of the code with any unnecessary complexity in interpreting the code.